enzyme/Enzyme/MLIR/Interfaces/GradientUtils.cpp

```diff
@@ -295,13 +296,22 @@ void mlir::enzyme::MGradientUtils::forceAugmentedReturns() {
 
 LogicalResult MGradientUtils::visitChild(Operation *op) {
   if (mode == DerivativeMode::ForwardMode) {
+    // In absence of a proper activity analysis, approximate it by treating any
+    // side effect-free operation producing constants as inactive.
+    if (auto iface = dyn_cast<MemoryEffectOpInterface>(op)) {
```

Comment:
For ease, can we push this into a isConstantOperation function within GradientUtils?

And then this simply becomes isConstantOperation && all(isConstantValue), without the memory effect check?